akka-cluster-typed/src/main/scala/akka/cluster/typed/internal/receptionist/ClusterReceptionist.scala

```diff
@@ -134,14 +138,27 @@ private[typed] object ClusterReceptionist extends ReceptionistBehaviorProvider {
         })
 
       def notifySubscribersFor(key: AbstractServiceKey, state: ServiceRegistry): Unit = {
-        val msg = ReceptionistMessages.Listing(key.asServiceKey, state.actorRefsFor(key))
+        def nodeIsDowned(entry: ClusterReceptionist.Entry): Boolean = {
+          cluster.state.members.exists(member ⇒
+            member.uniqueAddress == entry.uniqueAddress(setup.selfUniqueAddress) &&
+              member.status == MemberStatus.Down
```

Comment:
What I wanted to do was to avoid sending out a `Listing` with an actor-ref re-added just to then notify again when it gets removed. Could happen when a concurrent `Register` adds to a key on one node and downing on the leader removes for the same key.

You mean we would have to keep track of observerd `MemberRemoved` for that case to have a consistent behavior?

Reply:
> You mean we would have to keep track of observerd MemberRemoved for that case to have a consistent behavior?

You would also have to keep track of exactly what has been sent out to each subscriber to have a full solution, since new subscribers can arrive inbetween.

It becomes more complicated than it's worth in my opinion. I also don't see much bad consequences of double notifications since the notification isn't a delta (change) but carries the full new Listing.

Reply:
Ach. Yeah, I agree, didn't realize how complicated it was.